internal/common/conversion/schema_description.go

```diff
@@ -2,15 +2,83 @@ package conversion
 
 import (
 	"reflect"
+	"slices"
```

Comment:
nit: file is named `schema_description`, we could transition to a more general name like `schema_generation` now that this functionality is also present

Reply:
changed, thx